jira/client.py

```diff
@@ -803,7 +803,7 @@ def group_members(self, group):
         result = {}
         for user in r['users']['items']:
             result[user['name']] = {'fullname': user['displayName'], 'email': user.get('emailAddress', 'hidden'),
-                                    'active': user['active']}
+                                    'active': user['active'], 'key': user['key']}
```

Comment:
As per some of the comments in #281, we may want to maintain backwards compatibility. 

@ssbarnea What do you think about adding a block like.

```
if user.get('key'):
    result[user['name']].append('key',user['key'])

Reply:
Yep, I'm the one that brought up the backwards compatibility issue. :)

This change maintains backward compatibility for users of this library by keeping the map keyed by the same value (`user['name']`), so that anyone who's depending on that behavior isn't broken when this fix is merged and deployed.   (Others had proposed changing it to be `user['key']`, which would definitely be a breaking change for anyone who was using `user['name']`, since it was only available as the dict key.)

Reply:
Oh!  Are you saying we may need to *also* be backward compatible w/ older versions of the Jira API, where this value wasn't present?  I suppose we could do that. I don't know for a fact whether that field was ever added or removed. The [documentation](https://docs.atlassian.com/software/jira/docs/api/REST/7.6.1/#api/2/group-getGroup) doesn't even list the fields that that endpoint returns, so I suppose it's best to be on the safe side.

What about just doing:

~~~
result[user['name']] = {'fullname': ...
    ...
    'key': user.get('key')
}
~~~

It simplifies the code, and also makes it clear to users "Hey, there's this key value that you're not getting from your API endpoint." when they have a `user['key']` that `is None`.